scrapy/spiders/crawl.py

```diff
@@ -65,9 +67,19 @@ class CrawlSpider(Spider):
     def __init__(self, *a, **kw):
         super().__init__(*a, **kw)
         self._compile_rules()
+        cls = self.__class__
+        self._is_warned = False
+        is_overriden = method_is_overridden(cls, CrawlSpider, "_parse_response")
+        if is_overriden:
```

Comment:
We should maintain backward compatibility: if they indeed override `_parse_response`, we need to call it where needed

I think something like this may work:

```suggestion
        if is_overriden:
            self.parse_with_rules = self._parse_response
```